app/src/main/java/com/android/sample/ui/navigation/NavigationActions.kt

```diff
@@ -26,8 +30,12 @@ object Screen {
   const val ACHIEVEMENTS = "Achievements Screen"
   const val PREFERENCES = "Preferences Screen"
   const val SETTINGS = "Settings Screen"
+  const val BODY_WEIGHT_CREATION = "BodyWeightTraining Screen"
+  const val RUNNING_CREATION = "RunningTraining Screen"
+  const val YOGA_CREATION = "YogaTraining Screen"
   const val SESSIONSELECTION = "Session Selection Screen"
-  const val IMPORTORCREATE = "Import Or Create Screen"
+  const val IMPORTORCREATE_BODY_WEIGHT = "Import Or Create Body Weight Screen"
+  const val IMPORTORCREATE_YOGA = "Import Or Create Yoga Screen"
   const val TEST = "Test Screen"
```

Comment:
I see the problem, and adding Screens and Routes to differenctiate which type is called in the classed is a bit ugly.

Maybe there's a better way of doing it?

Reply:
Yes it is ugly, no I didn't find another way.